### PR TITLE
Add simple retry logic

### PR DESCRIPTION
**Pull request by TomPretty**
## What does this change?
Add some simple retry logic to the guest accout creation step (biggest cause of 500 errors from the API).

### CODE REVIEW

File: src/create-reminder-signup/lib/fetch-with-retry.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+): Promise<Response> {
+	let response = await fetcher();
+	let numTries = 1;
+
+	while (!response.ok && numTries < maxTries) {
+		response = await fetcher();
+		numTries++;
+	}
+	return response;
+}
```
Comment (TomPretty) [inline src/create-reminder-signup/lib/fetch-with-retry.ts:15]: @tomrf1 I'll add a quick test for this function if we're happy with the general approach here, but wanted to see what you thought first. I've gone super simple. There's a bit of duplication mostly due to keeping typescript happy

Comment (TomPretty) [reply src/create-reminder-signup/lib/fetch-with-retry.ts:15]: oh javascript actually has a `do...while` loop - this would be perfect!

Comment (TomPretty) [reply src/create-reminder-signup/lib/fetch-with-retry.ts:15]: ahhh no I don't think it is :(

Comment (tomrf1) [reply src/create-reminder-signup/lib/fetch-with-retry.ts:15]: Looks good. And if you enjoy recursion:
```
export const fetchWithRetry = (
    fetcher: () => Promise<Response>,
    maxRetries: number,
): Promise<Response> => fetcher().then(response => {
    if (!response.ok && maxRetries > 0) {
        return fetchWithRetry(fetcher, maxRetries-1);
    } else {
        return response;
    }
});
```

Comment (TomPretty) [reply src/create-reminder-signup/lib/fetch-with-retry.ts:15]: ooo I think the recursive version is too good to pass on

File: src/create-reminder-signup/lib/fetch-with-retry.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Response } from 'node-fetch';
+
+export async function fetchWithRetry(
+	fetcher: () => Promise<Response>,
+	maxTries: number,
+): Promise<Response> {
+	const response = await fetcher();
```
Comment (tomrf1) [inline src/create-reminder-signup/lib/fetch-with-retry.ts:7]: I'm being pedantic here, but calling it with `maxTries` of 0 will still make a request! Would it make sense to rename it as `maxRetries`, and the condition becomes `maxRetries > 0`?

Comment (TomPretty) [reply src/create-reminder-signup/lib/fetch-with-retry.ts:7]: No I love that level of pedantry! Yeah I agree 👍 I'll make that change now.

File: src/create-reminder-signup/lib/fetch-with-retry.test.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+		const numFails = 5;
+		const maxRetries = 1;
+		const expectedNumTries = 2;
+		const fetcher = getMockFetcher(numFails);
+
+		await fetchWithRetry(fetcher, maxRetries);
+
+		expect(fetcher).toHaveBeenCalledTimes(expectedNumTries);
+	});
+});
```
Comment (TomPretty) [inline src/create-reminder-signup/lib/fetch-with-retry.test.ts:26]: These might not be the most crucial tests ever, but I was quite happy to get a test in there to stop e.g an off by one error

File: src/create-reminder-signup/lib/identity.ts [hunk 4]
```diff
@@ -7,6 +7,9 @@ import {
 	success,
 } from '../../lib/identity';
 import { ReminderStage } from '../lambda/models';
+import { fetchWithRetry } from './fetch-with-retry';
+
+const IDENTITY_ACCOUNT_CREATION_MAX_RETRIES = 1;
```
Comment (TomPretty) [inline src/create-reminder-signup/lib/identity.ts:12]: We can obviously up this, but I was just curious how much a single retry would help

File: src/create-reminder-signup/lib/fetch-with-retry.test.ts (hunk 3 above)
Comment (tomrf1) [reply src/create-reminder-signup/lib/fetch-with-retry.test.ts:26]: 🥇

**Review by tomrf1 — APPROVED** [review]
great!

**Pull request merged**